pkg/controller/machine/controller.go

```diff
@@ -426,9 +429,17 @@ func (r *ReconcileMachine) setPhase(machine *machinev1.Machine, phase string, er
 		machine.Status.LastUpdated = &now
 		if phase == phaseFailed && errorMessage != "" {
 			machine.Status.ErrorMessage = &errorMessage
+			if machine.Annotations == nil {
+				machine.Annotations = map[string]string{}
+			}
+			machine.Annotations[MachineInstanceStateAnnotationName] = unknownInstanceState
```

Comment:
this might cause skew if the actuators set something in the provider.Status. We might consolidate to consolidate that.